tests/Domain/Integrations/Repositories/EloquentIntegrationRepositoryTest.php

```diff
@@ -219,4 +222,119 @@ public function test_it_can_delete_an_integration(): void
             'status' => IntegrationStatus::Deleted,
         ]);
     }
+
+    public function test_it_can_activate_with_a_coupon(): void
+    {
+        $couponId = uuid::uuid4();
+        $couponCode = '123';
+
+        $integrationId = Uuid::uuid4();
+
+        $coupon = new Coupon(
+            $couponId,
+            true,
+            null,
+            $couponCode
+        );
+
+        $couponRepository = new EloquentCouponRepository();
```

Comment:
To avoid dependencies on external repositories you can use `CouponModel::query()->insert([]);`